packages/next-auth/src/lib/actions.ts

```diff
@@ -120,16 +125,12 @@ export function createActionURL(
   h: Headers | ReturnType<typeof headers>,
   basePath?: string
 ): URL {
-  const envUrl = process.env.AUTH_URL ?? process.env.NEXTAUTH_URL
-  if (envUrl) {
-    const { origin, pathname } = new URL(envUrl)
-    const separator = pathname.endsWith("/") ? "" : "/"
-    return new URL(`${origin}${pathname}${separator}${action}`)
-  }
-  const host = h.get("x-forwarded-host") ?? h.get("host")
-  const protocol = h.get("x-forwarded-proto") === "http" ? "http" : "https"
-  // @ts-expect-error `basePath` value is default'ed to "/api/auth" in `setEnvDefaults`
-  const { origin, pathname } = new URL(basePath, `${protocol}://${host}`)
-  const separator = pathname.endsWith("/") ? "" : "/"
-  return new URL(`${origin}${pathname}${separator}${action}`)
+  return createActionURLCore(
```

Comment:
Let's use the function from coredirectly, and remove this function definition - so all the usages should be `createActionURL` 🙏

Reply:
done